### PR TITLE
refactor(NODE-4637): clean up async interval

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR does a bit of cleanup on the async interval using in SDAM monitoring.  Specifically it

- Moves the async interval into the monitor.ts file from the utils
- Move the associated unit tests 
- Converts monitor.test.js to Typescript.  This was necessary because the existing interval tests were in Typescript, so I could either convert them to JS or convert monitor.test.js to TS.  I chose the latter.
- converts the `makeInterruptibleAsyncInterval` factory function into a class
- renames some variables to more closely align with their actual purpose and renames the interval `SDAMMonitorInterval` (name can be workshopped).

To aid in review, moved the interval to monitor.ts (and the associated tests) in the first commit, and then tried to be granular with the changes from then on.  My intention was to put all the code reorganization changes in to the first commit, so the actual code changes that I made can be easily viewed in a diff.

##### Is there new documentation needed for these changes?

No.

#### What is the motivation for this change?

General code cleanup and better code organization.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/sdam/topology.ts [hunk 1]
```diff
@@ -386,7 +386,9 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
   }
```
Comment (baileympearson) [inline src/sdam/topology.ts:387]: The missing overloads in this file were revealed when I converted the monitor tests to TS.  I'm happy to revert these changes and cast in the test file instead.

Comment (nbbeeken) [reply src/sdam/topology.ts:387]: It's internal I think it's fine.

**Review by nbbeeken — APPROVED** [review]
LGTM, barring some bikeshedding on slack

**Review by nbbeeken — APPROVED** [review]
